lib/std/os.zig

```diff
@@ -6048,7 +6048,7 @@ pub const UnexpectedError = error{
 /// and you get an unexpected error.
 pub fn unexpectedErrno(err: E) UnexpectedError {
     if (unexpected_error_tracing) {
-        std.debug.print("unexpected errno: {d}\n", .{@intFromEnum(err)});
+        std.debug.print("unexpected error: E.{s}\n", .{@tagName(err)});
```

Comment:
`@tagName` is [checked illegal behavior on non-exhaustive enums](https://ziglang.org/documentation/master/#tagName), which errno seems to be on most (all?) platforms.
However, I believe that the default formatting for enums (so just using `{}` and passing `.{err}` directly) already has handling for printing the name if available, and the numeric code otherwise. Perhaps we could use that?

Reply:
Thanks for the correction.  I assumed it was exhaustive without checking.

I avoided using the default formatting of enums, since it prints the full type name; e.g. on linux x86_64 I got `os.linux.errno.generic.E.ACCES`.  I think the additional information is unnecessary.

A simple solution it to copy the code from `fmt` when formatting a non-exhaustive enum.

Reply:
I fixed the incorrect handling of @TagName, but I had to add a new function.

Does it make sense to use @setCold(true); in both `errorName` and `unexpectedErrno` functions?

Reply:
Isn't the new function just a copy `std.enums.tagName`?

Reply:
> Isn't the new function just a copy `std.enums.tagName`?

You are right; I was unaware of the std.enums namespace. Thanks!
The code of the `errorName` function has been adapted from `fmt.formatType`.  Maybe that code should be updated?

Unrelated: does `@intFromEnum` cache the result value?  In my code I called @intFromEnum only once.